### PR TITLE
Improve updater and tests logging

### PR DESCRIPTION
**Pull request by sechkova**
**Fixes issue #**: #1083

**Description of the changes being introduced by the pull request**:

- Add NullHandler() to the top-level logger
- Replace logging.exception calls in Updater()

**Please verify and check that the pull request fulfills the following
requirements**:

- [ ] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

### CODE REVIEW

**Comment by sechkova** [comment]
The proposed changes here are related to getting rid of the logging when it is not needed.
On the other hand, probably there's also a similar need of a common way to enable/increase the logging when needed (for debugging etc)

**Review by trishankatdatadog — APPROVED** [review]
💯 

Too much noise out-of-the-box right now

We can figure out configurable options later

File: tuf/client/updater.py [hunk 1]
```diff
@@ -1151,12 +1151,12 @@ def neither_403_nor_404(mirror_error):
           # Otherwise, reraise the error, because it is not a simple HTTP
           # error.
           if neither_403_nor_404(mirror_error):
-            logging.exception('Misc error for root version '+str(next_version))
+            logger.exception('Misc error for root version '+str(next_version))
```
Comment (joshuagl) [inline tuf/client/updater.py:1154]: 💯 excellent find. I've looked at this code numerous times and not noticed the typo here.

**Review by joshuagl — APPROVED** [review]
Thank you for finding and fixing this issue @sechkova! I'm not sure I completely agree with the current logging behaviour, but let's figure that out later.

File: tuf/client/updater.py (hunk 1 above)
Comment (trishankatdatadog) [reply tuf/client/updater.py:1154]: My bad, thanks everyone

**Pull request merged**